src/WinRT.Runtime/ActivationFactory.cs

```diff
@@ -0,0 +1,285 @@
+        public static WinRTModule Instance => _instance ?? MakeWinRTModule();
+
+        public unsafe WinRTModule()
+        {
+            IntPtr mtaCookie;
+            Marshal.ThrowExceptionForHR(Platform.CoIncrementMTAUsage(&mtaCookie));
+            _mtaCookie = mtaCookie;
+        }
+
+        public static unsafe (ObjectReference<I> obj, int hr) GetActivationFactory<I>(string runtimeClassId, Guid iid)
+        {
+            var module = Instance; // Ensure COM is initialized
+            IntPtr instancePtr = IntPtr.Zero;
+            try
+            {
+                MarshalString.Pinnable __runtimeClassId = new(runtimeClassId);
+                fixed (void* ___runtimeClassId = __runtimeClassId)
+                {
+                    int hr = Platform.RoGetActivationFactory(MarshalString.GetAbi(ref __runtimeClassId), &iid, &instancePtr);
+                    if (hr == 0)
+                    {
+                        var objRef = ObjectReference<I>.Attach(ref instancePtr);
+                        return (objRef, hr);
+                    }
+                    else
+                    {
+                        return (null, hr);
+                    }
+                }
+            }
+            finally
+            {
+                MarshalInspectable<object>.DisposeAbi(instancePtr);
+            }
+        }
+
+        ~WinRTModule()
+        {
+            Marshal.ThrowExceptionForHR(Platform.CoDecrementMTAUsage(_mtaCookie));
+        }
+    }
+
+#if EMBED
+    internal
+#else
+    public
+#endif
+    static class ActivationFactory
+    {
+        public static IObjectReference Get(string typeName)
+        {
+            // Prefer the RoGetActivationFactory HRESULT failure over the LoadLibrary/etc. failure
+            int hr;
+            ObjectReference<IActivationFactoryVftbl> factory;
+            (factory, hr) = WinRTModule.GetActivationFactory<IActivationFactoryVftbl>(typeName, InterfaceIIDs.IActivationFactory_IID);
+            if (factory != null)
+            {
+                return factory;
+            }
+
+            var moduleName = typeName;
+            while (true)
+            {
+                var lastSegment = moduleName.LastIndexOf(".", StringComparison.Ordinal);
+                if (lastSegment <= 0)
+                {
+                    Marshal.ThrowExceptionForHR(hr);
+                }
+                moduleName = moduleName.Remove(lastSegment);
+
+                DllModule module = null;
+                if (DllModule.TryLoad(moduleName + ".dll", out module))
+                {
+                    (factory, hr) = module.GetActivationFactory(typeName);
+                    if (factory != null)
+                    {
+                        return factory;
+                    }
+                }
+            }
+        }
+
+#if NET
+        public static IObjectReference Get(string typeName, Guid iid)
+#else
+        public static ObjectReference<I> Get<I>(string typeName, Guid iid)

```

Comment:
Note: this could potentially break if someone tried to run some NS2.0 projections against the .NET 6+ version of WinRT.Runtime, because the generic method would not exist there anymore. Do we care about that scenario?

Reply:
If so, I would recommend still leaving the non-generic overload, so the updated projections can use that, but also just leave the "old" generic overload as well for binary compatibility purposes. Maybe mark it obsolete too on .NET 6+.

Reply:
I'd imagine that would be problematic in other places as well?

Reply:
I brought that up because we've had the same issue in the Toolkit (we made the same mistake). You'd basically get code from old assemblies running against the new runtime throwing `MissingMethodException`. I'd recommend keeping both, but just making the generic one `[EditorBrowsable(Never)]` and also `[Obsolete]` on .NET 6+. If no new projection is runtime code is ever calling this one, it won't matter that it's there. And only NS2.0 will use it.

Reply:
We don't support our .NET Standard projections on .NET 6+ as there are other things that will probably break like ComWrappers.  But I am fine with going with either approach.

Reply:
Oh, well if that's the case then we can also just save code and leave this as is, even better 😄